### ISSUE TITLE
vmalert: improve /health state

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#3063]
**Is your feature request related to a problem? Please describe.**
Currently, vmalert will report /health as ok if process was able to start. But it doesn't check if configured datasource, notifier or remote write are available, or stopped being available after some time. Would be great if /health would account for this conditions.

**Describe the solution you'd like**
/health to verify if datasource is available and:
1. at least one notifier is available if alerting rules are configured
2. at least one remote write is available if recording rules are configured

**Describe alternatives you've considered**

vmalert already exposes such a state in form of metrics. So an alternative solution for this is to configure alerting rules
for vmalert.

**Comment by Howie59** [comment]
we check it by ping the addrs?
> 1. at least one notifier is available if alerting rules are configured
> 2. at least one remote write is available if recording rules are configured

**Comment by hagen1778** [comment]
Good point. I think, the implementation would require extending `datasource` and `notifier` interfaces to support new method `Ping` or `Health`. This new method will then be used by vmalert to verify if the component is available.

Afaik, VictoriaMetrics, Prometheus and Alertmanager all support health-check endpoints.

**Comment by f41gh7** [comment]
I think it's better to keep `/health` endpoint as is. It allows to track if vmalert is responsive in general and is ready to serve requests.

 If `health` endpoint would fail the probe, orchestrator have to restart it in attempt to "fix" it. And it could create unhealth loop for the vmalert.

 Also, `ping` endpoint is not part of well known prometheus compatible APIs. So it'd be problematic to check health of the datasource or remote write.

**Comment by makasim** [comment]
Agree with @f41gh7. Anything beyond a basic connectivity check on /health doesn't work well in practice. The best way is to rely on metrics and alerting rules.
